core/objects.go

```diff
@@ -225,7 +258,14 @@ func (ch Challenge) RecordsSane() bool {
 	switch ch.Type {
 	case ChallengeTypeHTTP01:
 		for _, rec := range ch.ValidationRecord {
-			if rec.URL == "" || rec.Hostname == "" || rec.Port == "" || rec.AddressUsed == nil ||
+			// We no longer store the hostname and port in the validation record
+			// for HTTP-01 challenges. Instead, we'll rehydrate it from the URL
+			// field.
+			rhp, err := TransformURLIntoHostnameAndPort(rec.URL)
+			if err != nil {
+				return false
+			}
+			if rec.URL == "" || rec.AddressUsed == nil || rhp.Hostname == "" || rhp.Port == "" ||
 				len(rec.AddressesResolved) == 0 {
```

Comment:
Like I mentioned above, if we're able to guarantee that this will error instead of returning "" for `host` or `port`, then we can shorted this conditional.

```suggestion
			// We no longer store the hostname and port in the validation record
			// for HTTP-01 challenges. Instead, we'll rehydrate it from the URL
			// field.
			_, _, err := rec.getHostPort()
			if err != nil {
				return false
			}
			if rec.URL == "" || rec.AddressUsed == nil || len(rec.AddressesResolved) == 0 {
				return false
			}
```